.mergify.yml

```
@@ -4,22 +4,51 @@ defaults:
     # mergify.io has removed bot_account from its free open source plan.
     comment:
     # bot_account: ceph-csi-bot # mergify[bot] will be commenting.
-    merge:
+    queue:
       # merge_bot_account: ceph-csi-bot #mergify[bot] will be merging prs.
       # update_bot_account: ceph-csi-bot #mergify will randomly pick and use
       # credentials of users with write access to repo to rebase prs.
+      name: default
       method: rebase
       rebase_fallback: merge
-      strict: smart
-      strict_method: rebase
-    rebase:
-    # bot_account: ceph-csi-bot # same as update_bot_account.
-
+      update_method: rebase
+queue_rules:
+  - name: default
+    conditions:
+      # Conditions to get out of the queue (= merged)
+      - label!=DNM
+      - base~=^(devel)|(release-.+)$
+      - "#approved-reviews-by>=2"
+      - "#changes-requested-reviews-by=0"
+      - "approved-reviews-by=@ceph/ceph-csi-contributors"
+      - "approved-reviews-by=@ceph/ceph-csi-maintainers"
+      - "status-success=codespell"
+      - "status-success=multi-arch-build"
+      - "status-success=go-test"
+      - "status-success=golangci-lint"
+      - "status-success=commitlint"
+      - "status-success=mod-check"
+      - "status-success=lint-extras"
+      - "status-success=ci/centos/k8s-e2e-external-storage/1.21"
+      - "status-success=ci/centos/k8s-e2e-external-storage/1.22"
+      - "status-success=ci/centos/mini-e2e-helm/k8s-1.20"
+      - "status-success=ci/centos/mini-e2e-helm/k8s-1.21"
+      - "status-success=ci/centos/mini-e2e-helm/k8s-1.22"
+      - "status-success=ci/centos/mini-e2e-helm/k8s-1.23"
+      - "status-success=ci/centos/mini-e2e/k8s-1.20"
+      - "status-success=ci/centos/mini-e2e/k8s-1.21"
+      - "status-success=ci/centos/mini-e2e/k8s-1.22"
+      - "status-success=ci/centos/mini-e2e/k8s-1.23"
+      - "status-success=ci/centos/upgrade-tests-cephfs"
+      - "status-success=ci/centos/upgrade-tests-rbd"
+      - "status-success=DCO"
```

```suggestion
      - or:
        - and:
      - "status-success=codespell"
      - "status-success=multi-arch-build"
      - "status-success=go-test"
      - "status-success=golangci-lint"
      - "status-success=commitlint"
      - "status-success=mod-check"
      - "status-success=lint-extras"
      - "status-success=ci/centos/k8s-e2e-external-storage/1.21"
      - "status-success=ci/centos/k8s-e2e-external-storage/1.22"
      - "status-success=ci/centos/mini-e2e-helm/k8s-1.20"
      - "status-success=ci/centos/mini-e2e-helm/k8s-1.21"
      - "status-success=ci/centos/mini-e2e-helm/k8s-1.22"
      - "status-success=ci/centos/mini-e2e-helm/k8s-1.23"
      - "status-success=ci/centos/mini-e2e/k8s-1.20"
      - "status-success=ci/centos/mini-e2e/k8s-1.21"
      - "status-success=ci/centos/mini-e2e/k8s-1.22"
      - "status-success=ci/centos/mini-e2e/k8s-1.23"
      - "status-success=ci/centos/upgrade-tests-cephfs"
      - "status-success=ci/centos/upgrade-tests-rbd"
      - "status-success=DCO"
```
Will not work for ci/centos branch, 
I'll push change into this pr in a min

---

ah.. yes, ci branch need changes. @Rakshith-R please feel free to push the changes :+1: